--- a/src/forkpty.c
+++ b/src/forkpty.c
@@ -36,7 +36,7 @@
         return -1;
     }
 
-    int pts = open(devname, O_RDWR | O_NOCTTY);
+    int pts = open(devname, O_RDWR);
     if (pts < 0)
     {
         return -1;
@@ -69,9 +69,12 @@
     if (pid == 0)
     {
         setsid();
-        if (ioctl(pts, TIOCSCTTY, (char *)NULL) == -1)
-            exit(-1);
+        close(pts);
+        pts = open(devname, O_RDWR);
+        if (pts < 0)
+        {
+            exit(-1);
+        }
 
         dup2(pts, STDIN_FILENO);
